frontend/packages/knative-plugin/src/components/add/EventSource.tsx

```diff
@@ -56,10 +62,17 @@ const EventSource: React.FC<Props> = ({
     data: {
       [typeEventSource.toLowerCase()]: getEventSourceData(typeEventSource.toLowerCase()),
     },
+    yamlData: !_.isEmpty(eventYaml) ? safeDump(eventYaml) : '',
+    editorType: EditorType.Form,
   };
 
   const createResources = (rawFormData: any): Promise<K8sResourceKind> => {
-    const knEventSourceResource = getEventSourcesDepResource(rawFormData);
+    let knEventSourceResource: K8sResourceKind;
+    if (isKnownEventSourceSelected(rawFormData.type)) {
+      knEventSourceResource = getEventSourcesDepResource(rawFormData);
+    } else {
+      knEventSourceResource = getEventSourcesResourceFromYAML(rawFormData);
```

Comment:
Do we need `getEventSourcesResourceFromYAML` ? 

```
if (isKnownEventSourceSelected(rawFormData.type)) {
      knEventSourceResource = getEventSourcesDepResource(rawFormData);
 } else {
     knEventSourceResource = safeLoad(rawFormData.chartValuesYAML);
}

```

Reply:
`getEventSourcesResourceFromYAML` adds labels and annotations to the yamldata and returns the updated yaml.
`safeLoad(rawFormData.yamlData)` would load the sample yaml for the event source, without labels and annotations
I updated the function `getEventSourcesDepResource` to handle both and moved the logic for handling yamldata and formdata to separate functions. PTAL